extension/src/popup/components/sendPayment/SendConfirm/TransactionDetails/index.tsx

```diff
@@ -184,23 +193,26 @@ export const TransactionDetails = ({ goBack }: { goBack: () => void }) => {
 
   // handles signing and submitting
   const handleSend = async () => {
-    const server = new StellarSdk.Server(networkDetails.networkUrl);
-
     try {
+      const server = new StellarSdk.Server(networkDetails.networkUrl);
       const transactionXDR = await server
         .loadAccount(publicKey)
-        .then((sourceAccount: Types.Account) => {
-          const transaction = new StellarSdk.TransactionBuilder(sourceAccount, {
+        .then((sourceAccount: Types.Account) =>
```

Comment:
ah, I can't believe I didn't notice this before. Would you mind using the `await` pattern here rather than `then`? We should be consistent in the codebase

Reply:
ah ya  good catch https://github.com/stellar/freighter/pull/554/commits/1e24f7f47c5066c4232d5cc4b36c7e34b8d67eee